[UE5.5] Merge pull request #429 from mcottontensor/esm_fix

mcottontensor opened the pull request:
# Backport

This will backport the following commits from `master` to `UE5.5`:
 - [Merge pull request #429 from mcottontensor/esm_fix](https://github.com/EpicGamesExt/PixelStreamingInfrastructure/pull/429)

<!--- Backport version: 9.6.1 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

lukehb commented on Frontend/library/src/Config/Config.ts:
```diff
@@ -742,7 +742,13 @@ export class Config {
+                optionSetting.options = existingOptions;
+            }
+            optionSetting.selected = settingValue;
         }
```
Few questions:

1. Which bit of code actually calls `setOptionSettingValue`?

2. Is it worth having a `setOrAddOptionSettingValue` - my concern is adding if missing is bit unexpected based on the name `setOptionSettingValue`?

Wdyt?

mcottontensor replied:
It's only used in two places. One for the preferred codec and one for the streamer ids. My logic for adding it in the existing behaviour is that if you're trying to set the selected value of an option selector, it doesn't really make sense for it to fail silently and not do anything.

lukehb reviewed (approved):
lgtm

mcottontensor committed:
- Fixed issue with esm and cjs builds not being correctly configured
- Adjusting how Protocol and Transport handle protocol BaseMessages. Transport should just handle strings. Protocol should handle messages.

(cherry picked from commit 84b4253f3077db4913c374c37bb1da6db5173939)

mcottontensor committed:
Fixing up logging from Common in signalling lib

(cherry picked from commit 60a8f2278f064ce34071ccb1c6736b8b20ef51f2)

mcottontensor committed:
Moved signalling protocol documentation

(cherry picked from commit 80aa060d880a8c194a04b83f18bd1ee360ab20e1)

mcottontensor committed:
Rebuilt common docs

(cherry picked from commit 709d6fe90630bd597d2e77e84a91c810a5896d5e)

mcottontensor committed:
Rebuilt signalling docs

(cherry picked from commit 127c7c63116e92d6bad27dff0ab3432470b4186c)

mcottontensor committed:
Frontend watch should use dev

(cherry picked from commit f004b846aac4abe02b57c10c17ebc02a078e1d8e)
